agent/utils/buffer-ring/bigqueue/bigqueue.go

```diff
@@ -155,25 +149,10 @@ func initPaths(dir string) error {
 	return nil
 }
 
-// fillInUse fills sizeInUse that represents the amount of space used in bigqueue with messages data.
-func (r *Ring) fillInUse() error {
-	for i := uint64(0); ; i += paginationSize {
-		data, _, err := r.fq.PeekPagination(i, paginationSize)
-		if err != nil {
-			return err
-		}
-		if len(data) == 0 {
-			return nil
-		}
-		for j := range data {
-			r.sizeInUse += uint32(len(data[j]))
-		}
-	}
-}
-
 // drain tries to drain cache.
-func (r *Ring) drain(amount uint32) error {
-	for size := uint32(0); size < amount; {
+func (r *Ring) drain(amount int64) error {
```

Comment:
we still have this problem because of an amount passed

1. New Ring 8 bytes
2. Push 7 bytes called first
3. following pushes called simultaneously
  a) Push 5 bytes
  b) Push 2 bytes
4. drain goroutine is called from step 3a: amount - 4 or 6 depending on the timing
5. drain goroutine is called from step 3b: amount - 1 or 6 depending on the timing
6. At this moment in time, the size of the ring might be 12, 7, or 0, depending on the operators' order.